### PR TITLE
Add support for Streaming Notifications

### PR DESCRIPTION
**Pull request by tkudiyarova**

### CODE REVIEW

File: lib/ews/connection.rb [hunk 1]
```diff
@@ -93,6 +93,13 @@ def post(xmldoc)
     check_response( @httpcli.post(@endpoint, xmldoc, headers) )
   end
 
+  # Send an asynchronous POST request to the web service
+  # @return HTTPClient::Connection instance
+  def post_async(xmldoc)
+    authenticate
```
Comment (wyaeld) [inline lib/ews/connection.rb]: why does `post_async` do `authenticate` when `post` does not?

File: lib/ews/soap/exchange_notification.rb [hunk 2]
```diff
@@ -102,6 +102,27 @@ def get_events(subscription_id, watermark)
...
+    # Used by streaming subscription clients to request notifications from the Client Access server
+    # @see https://msdn.microsoft.com/EN-US/library/ff406172(v=exchg.150).aspx
+    # @see https://msdn.microsoft.com/en-us/library/office/dn458792(v=exchg.150).aspx
+    #
+    # @param subscription_ids   [Array<String>] An array of Subscription identifiers
+    # @param connection_timeout [Fixnum] Specifies the number of minutes to keep a connection open
+    #        https://msdn.microsoft.com/EN-US/library/ff406137(v=exchg.150).aspx
+    def get_streaming_events(subscription_ids, connection_timeout = 30)
+      req = build_soap! do |type, builder|
+        if(type == :header)
```
Comment (wyaeld) [inline lib/ews/soap/exchange_notification.rb]: empty conditional?

**Comment by wyaeld** [comment]
Looks good, just a question on the authenticate

File: lib/ews/connection.rb (hunk 1 above)
Comment (tkudiyarova) [inline lib/ews/connection.rb]: I have `401 Unauthorized` error without `authenticate`.  Looks like `set_auth` method is not enough for async_post request, found this [related issue](https://github.com/nahi/httpclient/issues/181).

Comment (wyaeld) [inline lib/ews/connection.rb]: Ok, can you add those details and link in a comment on this code so the next person will be able to understand

**Comment by wyaeld** [comment]
LGTM, just needs comments

**Pull request merged**